realm/realm-library/src/androidTest/java/io/realm/TestHelper.java

```diff
@@ -46,6 +47,23 @@
 
 public class TestHelper {
 
+    public static class ExpectedCountCallback implements RealmCache.Callback {
```

Comment:
I don't agree. This change is good because then we can guarantee that the code runs within the proper locks. Otherwise you require the calling code to implement those locks, and then we end up where we are now.

It doesn't look particular nice though, that we can agree on :)